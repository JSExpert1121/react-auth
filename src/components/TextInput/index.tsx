// @material ui components
import TextField from '@material-ui/core/TextField';

// i18n
import authMsgs from 'translations/messages/page.auth';

// custom components
import withValidators from 'components/HOCs/withValidators';
import {
    isValidEmail,
    isRequired,
    isValidChar,
    isValidLengthMin,
    isValidLengthMax,
    isPasswordMatched
} from 'helper/validators';

export const EmailInput = withValidators([
    { validator: isRequired, message: authMsgs.emailRequired },
    { validator: isValidEmail, message: authMsgs.emailInvalid }
])(TextField);

export const NameInput = withValidators([
    { validator: isRequired, message: authMsgs.nameRequired }
])(TextField);

export const PassInput = withValidators([
    { validator: isRequired, message: authMsgs.passRequired },
    { validator: isValidChar, message: authMsgs.passInvalid },
])(TextField);

export const PassConfirmInput = withValidators([
    { validator: isRequired, message: authMsgs.passRequired },
    { validator: isValidChar, message: authMsgs.passInvalid },
    { validator: isValidLengthMin(8), message: authMsgs.passShort },
    { validator: isValidLengthMax(24), message: authMsgs.passLong },
    { validator: isPasswordMatched, message: authMsgs.passMismatch },
])(TextField);