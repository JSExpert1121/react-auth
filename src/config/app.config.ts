export const KAKAOMAP_API_KEY = '44a509cf4bf161e250c940454df5977c';
export const SUPPORTED_COUNTRIES = ['kr', 'jp', 'cn', 'hk', 'sg', 'tw', 'th', 'my', 'au', 'us', 'gb', 'de', 'fr', 'kz', 'ae', 'in'];
// export const COUNTRY_CODES = ['82', '81', '86', '852', '65', '886', '66', '60', '61', '1', '44', '49', '49', '33', '7', '971', '91'];
