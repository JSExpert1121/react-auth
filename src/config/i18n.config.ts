/**
 * @name i18n.config
 * @description config file for i18n. currently support ko and en
 * @version 1.0
 * @todo add locales
 * @author Darko
 */
const enMessages = require('translations/en.json');
const koMessages = require('translations/ko.json');

export const DEFAULT_LOCALE = 'en';

export const appLocales = [
    { code: 'en', name: 'English' },
    { code: 'ko', name: 'Korean' }
];

// auxilliary function to generate messages for each locales
export const formatMessages: (locale: string, messages: Record<string, string>) => Record<string, string> = (locale, messages) => {
    const defaultFormattedMessages =
        locale !== DEFAULT_LOCALE
            ? formatMessages(DEFAULT_LOCALE, enMessages)
            : {};
    const flattenFormattedMessages = (formattedMessages: Record<string, string>, key: string) => {
        const formattedMessage =
            !messages[key] && locale !== DEFAULT_LOCALE
                ? defaultFormattedMessages[key]
                : messages[key];
        return Object.assign(formattedMessages, { [key]: formattedMessage });
    };
    return Object.keys(messages).reduce(flattenFormattedMessages, {});
};

export const translationMessages = {
    en: formatMessages('en', enMessages),
    ko: formatMessages('ko', koMessages),
};
