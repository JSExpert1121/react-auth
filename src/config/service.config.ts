export const API_GATEWAY_ADDRESS = 'https://apig.t.moria.hitit.xyz:8765';
export const OAUTH_CLIENT_ID = 'web';
export const OAUTH_CLIENT_SECRET = 'hititweb42!';