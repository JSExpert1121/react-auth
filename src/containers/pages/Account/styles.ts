import { Theme, createStyles } from '@material-ui/core/styles';
import Colors from 'assets/colors';

export const styleFn = (theme: Theme) => createStyles({
    root: {
        width: '100%',
        padding: theme.spacing(3),
        maxWidth: 512,
        minWidth: 480,
    },
    form: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        width: '100%',
        textAlign: 'center'
    },
    logo: {
        padding: theme.spacing(2, 0)
    },
    title: {
        fontSize: '1.5rem',
        color: 'rgba(0, 0, 0, 0.8)',
        padding: theme.spacing(2)
    },
    text: {
        fontSize: '0.875rem',
        color: 'rgba(0, 0, 0, 0.7)',
        padding: theme.spacing(2)
    },
    success: {
        color: 'rgba(0, 0, 0, 0.9)',
        padding: theme.spacing(1)
    },
    error: {
        fontSize: '0.875rem',
        color: Colors.error,
        padding: theme.spacing(0, 0, 0, 2)
    },
    busy: {
        width: '100%',
    },
    submit: {
        margin: theme.spacing(2, 0),
        height: 'calc(100% - 12px)'
    },
    bottomMenu: {
        width: '100%',
        display: 'flex',
        padding: theme.spacing(2, 0)
    }
});