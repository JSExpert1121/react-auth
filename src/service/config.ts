import { oauthClient } from './base';

export const setLanguage = (lang: string) => {
    oauthClient.setLanguage(lang);
}
