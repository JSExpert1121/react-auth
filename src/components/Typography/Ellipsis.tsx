import React from 'react';
import { makeStyles, Theme } from '@material-ui/core/styles';

interface IEllipsisProps {
    fontSize?: string;
    maxLines?: number;
    hidden?: boolean;
}

const useStyles = (props: IEllipsisProps) => makeStyles((theme: Theme) => ({
    ellipsis: {
        display: '-webkit-box',
        maxHeight: `calc(${props.maxLines} * ${1.5} * ${props.fontSize})`,
        lineHeight: '1.5',
        visibility: !!props.hidden ? 'hidden' : 'visible',
        fontSize: props.fontSize,
        '-webkitLineClamp': props.maxLines,
        '-webkitBoxOrient': 'vertical',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
    }
}));

const EllipsisText: React.FC<IEllipsisProps> = (props) => {
    const { children, fontSize, maxLines, hidden } = props;
    let realSize = fontSize || '1rem';
    let realLines = maxLines || 2;

    const classes = useStyles({ fontSize: realSize, maxLines: realLines, hidden })({});
    return (
        <span className={classes.ellipsis}>
            {children}
        </span>
    );
};

export default EllipsisText;
