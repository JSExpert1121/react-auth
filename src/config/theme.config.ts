import { createMuiTheme } from '@material-ui/core/styles';

const theme = createMuiTheme({
	palette: {
		primary: {
			main: '#3f51b5',
			dark: '#303f9f',
		},
		secondary: {
			light: '#0033EE',
			main: '#1100BB',
			contrastText: '#000000',
		},
		background: {
			default: '#FFF'
		}
	},
	typography: {
		fontFamily: "\"Noto Sans\", \"Noto Sans KR\", sans-serif"
	},
	overrides: {
		MuiAppBar: {
			colorPrimary: {
				color: '#111',
				backgroundColor: 'white'
			}
		},
		MuiContainer: {
			root: {
				padding: 0
			}
		},
		MuiBadge: {
			colorSecondary: {
				color: 'white'
			}
		},
		MuiExpansionPanel: {
			root: {
				margin: '8px 0',
				'&:before': {
					height: 0
				}
			}
		},
		MuiExpansionPanelSummary: {
			root: {
				padding: 0,
				'&:before': {
					height: 0
				}
			},
			content: {
				margin: 0,
				flexGrow: 0,
				'&.Mui-expanded': {
					margin: 0
				}
			}
		},
		MuiExpansionPanelDetails: {
			root: {
				padding: 8
			}
		}
	}
});

export default theme;
