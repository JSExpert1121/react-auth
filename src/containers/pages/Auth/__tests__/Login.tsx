import React from 'react';
import { MemoryRouter, withRouter } from 'react-router-dom';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build';

import { createMount } from '@material-ui/core/test-utils';
import { ThemeProvider } from '@material-ui/styles';
import { IntlProvider, injectIntl } from "react-intl";
import { act } from 'react-dom/test-utils';

import Typography from '@material-ui/core/Typography';
import Link from '@material-ui/core/Link';
import Button from '@material-ui/core/Button';

import DefLoginPage, { LoginPage } from '../Login';
import { EmailInput, PassInput } from 'components/TextInput';

import { translationMessages } from 'config/i18n.config';
import theme from 'config/theme.config';
import messages from 'translations/messages/page.auth';

// store mock
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
import {
    GLOBAL_SET_TITLE,
    USER_LOGIN_SUCCESS
} from 'store/constants';
jest.mock('service/member');

configure({ adapter: new Adapter() });
const InjectedLoginPage = withRouter(injectIntl(LoginPage));
const ConnectedLoginPage = withRouter(DefLoginPage);


describe('containers/pages/auth/login', () => {

    const setTitle = jest.fn();
    const login = jest.fn();
    const showMsg = jest.fn();
    const hideMsg = jest.fn();
    login.mockResolvedValueOnce(undefined).mockRejectedValueOnce('Wrong credentials');

    // strange warning message ignore
    const mockConsoleMethod = (realConsoleMethod) => {
        const ignoredMessages = ['test was not wrapped in act(...)'];
        return (message, ...args) => {
            const containsMessage = ignoredMessages.some(ignoredMessage => message.includes(ignoredMessage))
            if (!containsMessage) {
                realConsoleMethod(message, ...args);
            }
        }
    }

    console.warn = jest.fn(mockConsoleMethod(console.warn));
    console.error = jest.fn(mockConsoleMethod(console.error));

    const mount = createMount();
    const wrapper = mount(
        <MemoryRouter>
            <IntlProvider locale='en' messages={translationMessages.en}>
                <ThemeProvider theme={theme}>
                    {<InjectedLoginPage setTitle={setTitle} login={login} showMessage={showMsg} hideMessage={hideMsg} />}
                </ThemeProvider>
            </IntlProvider>
        </MemoryRouter>
    );

    test('Snapshot test', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('UI components for english', () => {

        const texts = wrapper.find(Typography);
        expect(texts.at(0).props().children).toEqual('Log in with your Hit! T account');

        let control = wrapper.find(EmailInput);
        expect(control).toHaveLength(1);
        expect(control.props().label).toEqual('Email Address');
        expect(control.props().value).toEqual('');
        expect(control.props().message).toEqual(undefined);

        control = wrapper.find(PassInput);
        expect(control).toHaveLength(1);
        expect(control).toHaveLength(1);
        expect(control.props().label).toEqual('Password');
        expect(control.props().value).toEqual('');
        expect(control.props().message).toEqual(undefined);

        expect(wrapper.find(Link)).toHaveLength(2);
        expect(wrapper.find(Link).at(0).props().to).toEqual('/reset');
        expect(wrapper.find(Link).at(0).props().children).toEqual('Forgot password?');
        expect(wrapper.find(Link).at(1).props().to).toEqual('/signup');
        expect(wrapper.find(Link).at(1).props().children).toEqual("Don't have an account? Sign Up");

        expect(wrapper.find(Button)).toHaveLength(1);
        expect(wrapper.find(Button).at(0).props().children).toEqual('Sign in');
        expect(wrapper.find(Button).at(0).props().disabled).toEqual(false);
    });

    test('Email change event(invalid address)', done => {

        expect(setTitle).toHaveBeenCalledWith('Log in');
        act(() => {
            wrapper.find('#email').last().simulate('change', { target: { value: 'dad' } });
        });

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find(EmailInput).props().message).toEqual(messages.emailInvalid);
            expect(wrapper.find(Button).at(0).props().disabled).toEqual(true);
            done();
        });
    });

    test('Email change event(empty string)', done => {

        act(() => {
            wrapper.find('#email').last().simulate('change', { target: { value: '' } });
        });

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find(EmailInput).props().message).toEqual(messages.emailRequired);
            expect(wrapper.find(Button).at(0).props().disabled).toEqual(true);
            done();
        });
    });

    test('Email change event(valid address)', done => {

        act(() => {
            wrapper.find('#email').last().simulate('change', { target: { value: 'test@test.com' } });
        });

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find(EmailInput).props().message).toEqual(undefined);
            expect(wrapper.find(Button).at(0).props().disabled).toEqual(false);
            done();
        });
    });

    test('Password change event(empty)', done => {

        act(() => {
            wrapper.find('#password').last().simulate('change', { target: { value: '' } });
        });

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find(PassInput).props().message).toEqual(messages.passRequired);
            expect(wrapper.find(Button).at(0).props().disabled).toEqual(true);
            done();
        });
    });

    test('Password change event(valid)', done => {
        act(() => {
            wrapper.find('#password').last().simulate('change', { target: { value: '1234' } });
        });

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find(PassInput).props().message).toEqual(undefined);
            expect(wrapper.find(Button).at(0).props().disabled).toEqual(false);
            done();
        });
    });

    test('sign in button click event', done => {
        act(() => {
            wrapper.find(Button).simulate('click');
        });

        setImmediate(() => {
            wrapper.update();
            expect(login.mock.calls).toHaveLength(1);
            expect(login).toHaveBeenCalledWith('test@test.com', '1234');
            done();
        });
    });

    test('other key press on email input', done => {
        act(() => {
            wrapper.find(EmailInput).last().simulate('keypress', { key: '' });
        });

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find(PassInput).first().props().inputRef.current).not.toBe(document.activeElement);
            // expect(wrapper.find(PassInput).first().props().inputRef.current).toEqual(document.activeElement);
            done();
        });
    });

    test('enter key press on email input', done => {
        act(() => {
            wrapper.find(EmailInput).last().simulate('keypress', { key: 'Enter' });
        });

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find(PassInput).first().props().inputRef.current).toEqual(document.activeElement);
            done();
        });
    });

    test('enter key press on password input', done => {
        act(() => {
            wrapper.find(PassInput).last().simulate('keypress', { key: 'Enter' });
        });

        setImmediate(() => {
            wrapper.update();
            expect(login.mock.calls).toHaveLength(2);
            expect(login).lastCalledWith('test@test.com', '1234');
            done();
        });
    });

    test('other key press on password input', done => {
        act(() => {
            wrapper.find(PassInput).last().simulate('keypress', { key: '' });
        });

        setImmediate(() => {
            wrapper.update();
            expect(login.mock.calls).toHaveLength(2);
            done();
        });
    });
});

describe('connected component', () => {

    const mount = createMount();

    const username = 'user1@test.com';
    const password = 'testuser1';

    const store = mockStore({
        user: undefined,
        token: undefined
    });

    const wrapper = mount(
        <Provider store={store}>
            <MemoryRouter>
                <IntlProvider locale='en' messages={translationMessages.en}>
                    <ThemeProvider theme={theme}>
                        {<ConnectedLoginPage />}
                    </ThemeProvider>
                </IntlProvider>
            </MemoryRouter>
        </Provider>
    );

    test('setTitle', done => {
        wrapper.update();
        expect(store.getActions()).toHaveLength(1);
        expect(store.getActions()).toContainEqual({
            type: GLOBAL_SET_TITLE,
            payload: 'Log in'
        });
        done();
    });

    test('Email change event(valid address)', done => {

        act(() => {
            wrapper.find('#email').last().simulate('change', { target: { value: username } });
        });

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find(EmailInput).props().message).toEqual(undefined);
            expect(wrapper.find(Button).at(0).props().disabled).toEqual(false);
            done();
        });
    });

    test('Password change event(valid)', done => {
        act(() => {
            wrapper.find('#password').last().simulate('change', { target: { value: password } });
        });

        setImmediate(() => {
            wrapper.update();
            expect(wrapper.find(PassInput).props().message).toEqual(undefined);
            expect(wrapper.find(Button).at(0).props().disabled).toEqual(false);
            done();
        });
    });

    test('login', async done => {
        await wrapper.find(Button).first().props().onClick(undefined);
        expect(store.getActions()).toContainEqual({
            type: USER_LOGIN_SUCCESS,
            payload: `${username}__private_token`
        });
        done();
    });
})