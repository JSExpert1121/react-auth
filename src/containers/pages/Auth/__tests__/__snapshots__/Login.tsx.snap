// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/pages/auth/login Snapshot test 1`] = `ReactWrapper {}`;
